app/src/main/java/com/example/android/oscarine/signin/SigninFragment.kt

```diff
@@ -12,20 +12,24 @@ import com.example.android.oscarine.databinding.FragmentSigninBinding
 
 class SigninFragment : Fragment() {
 
-    private val viewModel: SigninViewModel by lazy {
-        ViewModelProviders.of(this).get(SigninViewModel::class.java)
-    }
+    private lateinit var viewModel: SigninViewModel
+    private lateinit var viewModelFactory: SigninViewModelFactory
 
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                               savedInstanceState: Bundle?): View? {
         val binding = FragmentSigninBinding.inflate(inflater)
 
+        viewModelFactory = SigninViewModelFactory(context!!)
```

Comment:
Inject SharedPreference instead of context

Reply:
I need context for using it in `SharedPreference` class

Reply:
Exactly, if you inject SharedPreference instead, you won't need it